Bdd/features/environment.py

```
@@ -33,6 +47,44 @@ def wait_for_tcp_port_open(host="syslog-ng", port=5514, timeout=5):
+    """Start a fresh otelcol-contrib (Windows) with the BDD config and wait
+    for the TCP/UDP listeners to bind. Mirrors what the CI workflow does
+    on first start (otelcol-contrib.exe + Bdd/otel/config.yaml, stdout/err
+    appended to Bdd/output/otelcol.{out,err}). Closes the parent's
+    stdout/err handles after Popen so they don't leak — the child keeps
+    its own duplicated copies. Uses os.path.join so the executable path
+    has Windows-native backslashes — _winapi.CreateProcess does not
+    resolve forward slashes the way bash does.
+    """
+    os.makedirs(os.path.dirname(OTELCOL_OUT), exist_ok=True)
+    with open(OTELCOL_OUT, "ab") as out, open(OTELCOL_ERR, "ab") as err:
+        subprocess.Popen(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)